src/LookupArrays/selector.jl

```diff
@@ -103,16 +103,30 @@ function at(lookup::NoLookup, sel::At; kw...)
     return r
 end
 function at(lookup::LookupArray, sel::At; kw...)
-    at(order(lookup), lookup, val(sel), atol(sel), rtol(sel); kw...)
+    at(order(lookup), span(lookup), lookup, val(sel), atol(sel), rtol(sel); kw...)
 end
 function at(
-    ::Ordered, lookup::LookupArray{<:Union{Number,Dates.TimeType}}, selval, atol, rtol::Nothing;
+    ::Ordered, span::Regular, lookup::LookupArray{<:Integer}, selval, atol::Nothing, rtol::Nothing;
+    err=_True()
+)
+    x = unwrap(selval)
+    Δ = step(span)
+    i, remainder = divrem(x - first(lookup), Δ)
+    i += firstindex(lookup)
+    if remainder == 0 && checkbounds(Bool, lookup, i)
+        return i
+    else
+        return _selnotfound_or_nothing(err, lookup, selval)
+    end
+end
+function at(
+    ::Ordered, ::Span, lookup::LookupArray{<:Union{Number,Dates.TimeType}}, selval, atol, rtol::Nothing;
     err=_True()
 )
     x = unwrap(selval)
     i = searchsortedlast(lookup, x)
     # Try the current index
-    if i === 0
+    if i == firstindex(lookup)
         i1 = i + 1
```

Comment:
```suggestion
        i1 = i + firstindex(lookup)
```

This is probably what you need to do